src/Digdir.Library.Utils.AspNet/HealthCheckExtensions.cs

```diff
@@ -0,0 +1,78 @@
+using Digdir.Library.Utils.AspNet.HealthChecks;
+using HealthChecks.UI.Client;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Diagnostics.HealthChecks;
+
+namespace Digdir.Library.Utils.AspNet;
+
+public static class HealthCheckExtensions
+{
+    public class AspNetHealthChecksOptions
+    {
+        public string WellKnownEndpointsConfigurationSectionPath { get; set; } = string.Empty;
+    }
+
+    public class JwtBearerTokenSchema
+    {
+        public string Name { get; set; } = string.Empty;
+        public string WellKnown { get; set; } = string.Empty;
+    }
+
+    public static IServiceCollection AddAspNetHealthChecks(this IServiceCollection services, IConfiguration configuration, Action<AspNetHealthChecksOptions> configure)
```

Comment:
Her må konfigurasjon registreringen og ansvarsfordelingen masseres litt mer. Vi tar inn IConfiguration **og** en configure action for en spesifikk konfigurajson. Mye på rett spor, men disse pleier å ha en XOR relasjon i slike kontekster. Personlig liker jeg at pakker har en nullable configure action fremfor å ta inn hele IConfiguration. Det er fordi pakker fort får litt for mye "power" med hele IConfiguration og skjuler de _faktiske_ avhengighetene for konsumenten av pakken. Hvilken _del_ av konfigurasjonen konsumeres av pakken? Hvilken del er påkrevd? Hvilken del er optional? Mye implisitt. Det er selvfølgelig unntak til denne "regelen". Men det er mer forbeholdt store bibliotek med god config dokumentasjon Eks serilog. 

En nullable configure action gjør at konsumenter kan _gi eksplisitt_ konfigurasjon til pakker, enten via configure action eller via interfacene [IConfigureOptions<TOptions>](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.options.iconfigureoptions-1), [IPostConfigureOptions<TOptions>](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.options.ipostconfigureoptions-1), eller deres "named" varianter. 

Se [Options pattern guidance for .NET library authors](https://learn.microsoft.com/en-us/dotnet/core/extensions/options-library-authors) hvis it tickles your fancy. 

Det vi faktisk er interessert i av konfigurasjon her er kun well known url endepunktene. Men av det jeg forstår av koden er det verken påkrevd, eller spesifikk til well known endepunkt - kun http get endepunkt. Dette kan vi eksponere til konsumenten via en settings objekt med en liste av string eller urler:
```csharp
public class AspNetHealthChecksSettings
{
    public List<string>? HttpGetEndpointsToCheck { get; set; }
}
```

Men har ikke altinn autorisasjon sin egen health check som vi kunne sjekke i stede for å sjekke spesifikke get endepunkt?

Reply:
Hører med ulike teams her om de har et eget health-check-endpoint

Reply:
>En nullable configure action gjør at konsumenter kan gi eksplisitt konfigurasjon til pakker, enten via configure action eller via interfacene [IConfigureOptions](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.options.iconfigureoptions-1), [IPostConfigureOptions](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.options.ipostconfigureoptions-1), eller deres "named" varianter.

Agree!